cache/refs.go

```diff
@@ -594,8 +596,14 @@ func (sr *immutableRef) ociDesc(ctx context.Context, dhs DescHandlers) (ocispecs
 	desc := ocispecs.Descriptor{
 		Digest:      sr.getBlob(),
 		Size:        sr.getBlobSize(),
-		MediaType:   sr.getMediaType(),
 		Annotations: make(map[string]string),
+		MediaType:   sr.getMediaType(),
+	}
+
+	if convertNonDist {
+		desc.MediaType = convert.LayerToDistributable(desc.MediaType)
```

Comment:
It looks like this is the only place where this function is called. So it can be a private function in the same package.

Reply:
Fixed